File: blog-admin/src/api/article.ts
import request from '@/utils/request'

export interface ArticleLikeRankItem {
  articleId: number
  title: string
  likeCount: number
  rank: number
}

export interface ArticleRecord {
  id?: number
  title: string
  summary?: string
  content?: string
  viewCount?: number
  likeCount?: number
  commentCount?: number
  publishTime?: string
  status?: number
  authorName?: string
}

export interface PageResult<T> {
  total: number
  current: number
  size: number
  records: T[]
}

export const articleApi = {
  ranking(limit = 10) {
    return request.get<ArticleLikeRankItem[]>('/admin/article/ranking', { params: { limit } })
  },
  page(params: { current: number; size: number; title?: string; author?: string; status?: number }) {
    return request.get<PageResult<ArticleRecord>>('/admin/article/page', { params })
  },
  updateStatus(id: number, status: number) {
    return request.put<unknown>(`/admin/article/${id}/status`, undefined, { params: { status } })
  },
  getById(id: number) {
    return request.get<ArticleRecord>(`/admin/article/${id}`)
  },
  getTagIds(id: number) {
    return request.get<number[]>(`/admin/article/${id}/tags`)
  },
  create(data: any) {
    return request.post<unknown>('/admin/article', data)
  },
  update(id: number, data: any) {
    return request.put<unknown>(`/admin/article/${id}`, data)
  },
}

File: blog-android/app/src/main/java/cn/yzfy/blog/data/net/dto/Article.java
package cn.yzfy.blog.data.net.dto;

/**
 * @author 一朝风月
 * @description 文章 DTO（portal/article 相关接口）。
 * @datetime 2026-03-18
 */
public class Article {
    public Long id;
    public Long userId;
    public String title;
    public String summary;
    public String content;
    public String coverImage;
    public Integer status;
    public Integer topFlag;
    public Long viewCount;
    public String publishTime;
    public String createTime;
    public String updateTime;
}

